src/pybind/mgr/cephadm/module.py

```diff
@@ -2015,6 +2040,56 @@ def update_rbd_mirror(self, spec):
         spec = NodeAssignment(spec=spec, get_hosts_func=self._get_hosts, service_type='rbd-mirror').load()
         return self._update_service('rbd-mirror', self.add_rbd_mirror, spec)
 
+    def add_prometheus(self, spec):
+        if not spec.placement.hosts or len(spec.placement.hosts) < spec.count:
+            raise RuntimeError("must specify at least %d hosts" % spec.count)
```

Comment:
1. This deserves it's own function: `def placement_has_hosts(placement: PlacementSpec)`
2. https://docs.ceph.com/docs/master/mgr/orchestrator_modules/#orchestrator.OrchestratorValidationError

Reply:
@jschmid1 do we need this at all with the scheduler?